src/base/staticClient.ts

```diff
@@ -124,46 +128,35 @@ async function runFetch(
     options.queryParameters
   ).toString();
 
-  // Lets grab all the RequestInit defaults from the clientConfig
-  const defaultsFromClientConfig: RequestInit = {
-    cacheManager: options.client.clientConfig.cacheManager,
-    retry: options.client.clientConfig.retrySettings,
-  };
+  // Multiple headers can be specified by using different cases. The `Headers`
+  // class handles this automatically.
+  const headers = new Headers(options.client.clientConfig.headers);
+  for (const [header, value] of new Headers(options.headers)) {
+    // Headers specified on the request will _replace_ those specified on the
+    // client, rather than be appended to them.
+    headers.set(header, value);
+  }
 
-  // Let's create a request init object of all configurations in the current request
-  const currentOptionsFromRequest: RequestInit = {
+  const fetchOptions: fetch.FetchOptions = {
```

Comment:
some tests for these changes should be great

Reply:
Added.